### PR TITLE
`flake8` and `black`: Fix Problems and add Actions for verification

### PR DESCRIPTION
**Pull request by SmithChart**
This PR adds `flake8` and `black` to this project.

We first set the tooling up for local use. Running `make qa` will run both tools and output any problems found.

Next:
* The code is reformatted using `black` and some odd whitespaces removed (while we are on it).
* Findings by `flake8` are corrected and the imports re-sorted.

Finally this PR adds Github actions to run the tools on push and in PRs.

### CODE REVIEW

File: .github/workflows/flake8.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  flake8_py3:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.9
+          architecture: x64
+      - name: Checkout PyTorch
+        uses: actions/checkout@master
```
Comment (jluebbe) [inline .github/workflows/flake8.yml:18]:
```suggestion
      - uses: actions/checkout@v3
```

**Review by jluebbe — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -30,3 +31,18 @@ clean:
 	rm -rf $(PYTHON_ENV_ROOT)
 
 envs: env packaging-env
+
+# testing #####################################################################
+$(PYTHON_TESTING_ENV)/.created: REQUIREMENTS.qa.txt
+	rm -rf $(PYTHON_TESTING_DEV) && \
```
Comment (hnez) [inline Makefile:37]: It looks to me like `PYTHON_TESTING_DEV` is undefined here.

File: .github/workflows/black.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: black
+
+on:
```
Comment (hnez) [inline .github/workflows/black.yml:3]: For the `tacd`we have `on: pull_request` **and** `on: push` in our [workflow `.yaml`s](https://github.com/linux-automation/tacd/blob/main/.github/workflows/tacd-daemon.yaml). I think that should be the same here?

File: .github/workflows/flake8.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Install flake8
+        run: pip install flake8 flake8-bugbear flake8-pyproject
```
Comment (hnez) [inline .github/workflows/flake8.yml:20]: Maybe `pip install -r ./REQUIREMENTS.qa.txt` here to de-duplicate the QA dependencies?

File: Makefile (hunk 2 above)
Comment (SmithChart) [reply Makefile:37]: You are right. (This typo must be in different projects by now, as the initial commit goes back to 2021 :grinning: )

File: .github/workflows/black.yml (hunk 3 above)
Comment (SmithChart) [reply .github/workflows/black.yml:3]: LGTM. I've changed this here.

File: .github/workflows/flake8.yml (hunk 4 above)
Comment (SmithChart) [reply .github/workflows/flake8.yml]: In this case we need to checkout the repo first. I've added that step and now install the `REQUIREMENTS.qa.txt`

**Comment by SmithChart** [comment]
@hnez @jluebbe feel free to have another look.

**Review by hnez — CHANGES_REQUESTED** [review]
The rest looks good to me.

File: .github/workflows/black.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  push:
+    paths:
+      - '*.py'
+  pull_request:
+    paths:
+      - '*.py'
```
[the remark this answers is not in the document]
Comment (SmithChart) [reply .github/workflows/black.yml]: Updated to `**.py`. But I did not see the action running on this PR. Not sure why that is.

File: .github/workflows/flake8.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+name: flake8 
+
+on:
+  push:
+    paths:
+      - '*.py'
+  pull_request:
+    paths:
+      - '*.py'
```
[the remark this answers is not in the document]
Comment (SmithChart) [reply .github/workflows/flake8.yml]: Updated to `**.py`. But I did not see the action running on this PR. Not sure why that is.

**Review by hnez — APPROVED** [review]
I'm still a bit confused as to why the jobs are not executed for this PR (as I think that's usually the case for PRs that add jobs) I would however be okay with merging it as-is and maybe fixing whatever prevents the jobs from running in a follow-up PR.

**Review by hnez — APPROVED** [review]

**Pull request merged**